src/lib/logical_query_plan/compare_lqps.cpp

```diff
@@ -0,0 +1,291 @@
+#include "compare_lqps.hpp"
+
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/aggregate_node.hpp"
+#include "logical_query_plan/create_view_node.hpp"
+#include "logical_query_plan/delete_node.hpp"
+#include "logical_query_plan/drop_view_node.hpp"
+#include "logical_query_plan/dummy_table_node.hpp"
+#include "logical_query_plan/insert_node.hpp"
+#include "logical_query_plan/join_node.hpp"
+#include "logical_query_plan/limit_node.hpp"
+#include "logical_query_plan/logical_plan_root_node.hpp"
+#include "logical_query_plan/mock_node.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/projection_node.hpp"
+#include "logical_query_plan/show_columns_node.hpp"
+#include "logical_query_plan/show_tables_node.hpp"
+#include "logical_query_plan/sort_node.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+#include "logical_query_plan/union_node.hpp"
+#include "logical_query_plan/update_node.hpp"
+#include "logical_query_plan/validate_node.hpp"
+
+namespace opossum {
+
+SemanticLQPCompare::SemanticLQPCompare(const std::shared_ptr<const AbstractLQPNode>& lhs, const std::shared_ptr<const AbstractLQPNode>& rhs): _lhs(lhs), _rhs(rhs) {
```

Comment:
c) This looks a lot like you are trying to implement `operator==` without naming it as such. What is the difference?

Reply:
The difference is, that I wouldn't be sure what the meaning of an `operator==` on the node of a Graph structure is. Does it just compare the nodes or does it also compare the children? That's why I'd prefer `lqp_equals` as the graph comparing function, since it clearly expresses that we're comparing entire plans. 

For comparing two nodes I'd prefer `NodeType::equals` over `NodeType::operator==` because of gut feeling - but that might be just me, I can live with both.